core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1077,7 +1077,7 @@ class ReplicaManager(val config: KafkaConfig,
           // Try the read first, this tells us whether we need all of adjustedFetchSize for this partition
           val readInfo: LogReadInfo = partition.readRecords(
             fetchOffset = fetchInfo.fetchOffset,
-            currentLeaderEpoch = fetchInfo.currentLeaderEpoch,
+            currentLeaderEpoch = toScalaOption(fetchInfo.currentLeaderEpoch),
```

Comment:
I did not realize that changing to a Scala Option would be that pervasive when I suggested it. It may be better to keep your original code that used a Java Option and refactor this once #9008 is completed.

Reply:
I think it's ok to keep the Scala Option here. https://github.com/apache/kafka/pull/9008 can just update the field if it changes name.

Reply:
Btw, you can use `.asScala` on a Java Option to convert it directly.

Reply:
Yes but it give us a `Option[java.lang.Integer]` oject while we want `Option[Int]`